test/cleanup.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# ------------------------------------------------------------------
+# [korthout] Removes the test repo, sourced from setup.sh
+# ------------------------------------------------------------------
+
+set -e  # Abort script at first error, when a command exits with non-zero status (except in until or while loops, if-tests, list constructs)
+# set -u  # Attempt to use undefined variable outputs error message, and forces an exit
+# set -x  # Similar to verbose mode (-v), but expands commands
+# set -o pipefail  # Causes a pipeline to return the exit status of the last command in the pipe that returned a non-zero return value.
+
+# ------------------------------------------------------------------
+# Lock script, to make sure it's only run once
+# ------------------------------------------------------------------
+
+SUBJECT=cleanup-test-repo
+LOCK_FILE=/tmp/${SUBJECT}.lock
+
+if [ -f "$LOCK_FILE" ]; then
+echo "Script is already running"
+exit
+fi
+
+trap 'rm -f $LOCK_FILE' EXIT
+touch $LOCK_FILE 
+
+# -----------------------------------------------------------------
+# Script
+# -----------------------------------------------------------------
+
+REPOSITORY_NAME="repo"
+PWDO="$(pwd)"
+
+cleanup () {
```

Comment:
suggestion:

the cleanup script could also reset the GitHub config. (The setup script could copy the old config to a safe place and then the cleanup copies the config back). Then you don't need to worry where you run the tests.

Reply:
I like this idea, but it's not necessary. The setup only defines local git config inside the newly created `repo` folder (just after the `git init`. The cleanup.sh deletes the folder and so also removes the local git config. The global is not changed by the test setup (but may be changed by the backport.sh production code script if necessary, note that it does not define local/global so it looks for whatever is available).